src/state/slices/actionSlice/selectors.ts

```diff
@@ -0,0 +1,62 @@
+import { selectEnabledWalletAccountIds } from '../common-selectors'
+import { swapSlice } from '../swapSlice/swapSlice'
+import { actionSlice } from './actionSlice'
+import { ActionStatus, isPendingSwapAction, isSwapAction } from './types'
+
+import { createDeepEqualOutputSelector } from '@/state/selector-utils'
+import { selectSwapIdParamFromFilter } from '@/state/selectors'
+
+export const selectWalletActions = createDeepEqualOutputSelector(
+  actionSlice.selectors.selectActions,
+  selectEnabledWalletAccountIds,
+  swapSlice.selectors.selectSwapsById,
+  (actions, enabledWalletAccountIds, swapsById) => {
+    return actions.filter(action => {
+      if (!isSwapAction(action)) return action
+
+      const swapId = action.swapMetadata.swapId
+      const relatedSwap = swapsById[swapId]
+
+      if (!relatedSwap?.sellAccountId) return false
+
+      return enabledWalletAccountIds.includes(relatedSwap.sellAccountId)
+    })
+  },
+)
+
+export const selectInitializedActionsByUpdatedAtDesc = createDeepEqualOutputSelector(
+  selectWalletActions,
+  actions => {
+    return actions.sort((a, b) => b.updatedAt - a.updatedAt)
+  },
+)
+
+export const selectWalletHasPendingActions = createDeepEqualOutputSelector(
+  selectWalletActions,
+  actions => {
+    return actions.filter(action => action.status === ActionStatus.Pending).length > 0
+  },
+)
+
+export const selectPendingSwapActions = createDeepEqualOutputSelector(
+  selectWalletActions,
+  actions => {
+    return actions.filter(isPendingSwapAction)
```

Comment:
nitpick: would find it more declarative to only check for `isSwapAction` for narrowing purposes and then a right-side `action.status === ActionStatus.Pending)` instead of declaring `isPendingSwapAction`, especially since this is only used in one place

Reply:
Actually reverted because TS doesn't like it at all